runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/collections/CaseInsensitiveMutableStringSet.kt

```diff
@@ -0,0 +1,40 @@
+package aws.smithy.kotlin.runtime.collections
+
+internal class CaseInsensitiveMutableStringSet(
+    initialValues: Iterable<CaseInsensitiveString> = setOf(),
+) : MutableSet<String> {
+    private val delegate = initialValues.toMutableSet()
+
+    override fun add(element: String) = delegate.add(element.toInsensitive())
+    override fun clear() = delegate.clear()
+    override fun contains(element: String) = delegate.contains(element.toInsensitive())
+    override fun containsAll(elements: Collection<String>) = elements.all { it in this }
+    override fun equals(other: Any?) = other is CaseInsensitiveMutableStringSet && delegate == other.delegate
+    override fun hashCode() = delegate.hashCode()
+    override fun isEmpty() = delegate.isEmpty()
+    override fun remove(element: String) = delegate.remove(element.toInsensitive())
+    override val size: Int get() = delegate.size
+    override fun toString() = delegate.toString()
+
+    override fun addAll(elements: Collection<String>) =
+        elements.fold(false) { modified, item -> add(item) || modified }
```

Comment:
Could be simplified to `elements.any { add(it) }`, same applies for `removeAll`

Reply:
No, [`any` short-circuits once it finds a match](https://github.com/JetBrains/kotlin/blob/c6f337283d59fcede75954eebaa589ad1b479aea/libraries/stdlib/common/src/generated/_Collections.kt#L1749-L1758). If we used that, we'd only ever add at most a single item.

Reply:
Oh ok. I updated the implementation before posting this comment and all the tests still passed, but it's probably just the way they're written